command/resource/delete/delete_test.go

```diff
@@ -0,0 +1,138 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+package delete
+
+import (
+	"testing"
+
+	"github.com/mitchellh/cli"
+	"github.com/stretchr/testify/require"
+
+	"github.com/hashicorp/consul/agent"
+	"github.com/hashicorp/consul/command/resource/apply"
+	"github.com/hashicorp/consul/testrpc"
+)
+
+func TestResourceDeleteInvalidArgs(t *testing.T) {
+	t.Parallel()
+
+	type tc struct {
+		args           []string
+		expectedCode   int
+		expectedErrMsg string
+	}
+
+	cases := map[string]tc{
+		"nil args": {
+			args:           nil,
+			expectedCode:   1,
+			expectedErrMsg: "Please provide required arguments",
+		},
+		"empty args": {
+			args:           []string{},
+			expectedCode:   1,
+			expectedErrMsg: "Please provide required arguments",
+		},
+		"missing file path": {
+			args:           []string{"-f"},
+			expectedCode:   1,
+			expectedErrMsg: "Failed to parse args: flag needs an argument: -f",
+		},
+		"provide type and name": {
+			args:           []string{"a.b.c"},
+			expectedCode:   1,
+			expectedErrMsg: "Must specify two arguments: resource type and resource name",
+		},
+		"does not provide resource name after type": {
+			args:           []string{"a.b.c", "-namespace", "default"},
+			expectedCode:   1,
+			expectedErrMsg: "Must provide resource name right after type",
+		},
+	}
+
+	for desc, tc := range cases {
+		t.Run(desc, func(t *testing.T) {
+			ui := cli.NewMockUi()
+			c := New(ui)
+
+			require.Equal(t, tc.expectedCode, c.Run(tc.args))
+			require.Contains(t, ui.ErrorWriter.String(), tc.expectedErrMsg)
+		})
+	}
+}
+
+func createResource(t *testing.T, a *agent.TestAgent) {
+	applyUi := cli.NewMockUi()
+	applyCmd := apply.New(applyUi)
+
+	args := []string{
+		"-http-addr=" + a.HTTPAddr(),
+		"-token=root",
+	}
+
+	args = append(args, []string{"-f=../testdata/demo.hcl"}...)
+
+	code := applyCmd.Run(args)
+	require.Equal(t, 0, code)
+	require.Empty(t, applyUi.ErrorWriter.String())
+}
+
+func TestResourceDelete(t *testing.T) {
+	if testing.Short() {
+		t.Skip("too slow for testing.Short")
+	}
+
+	t.Parallel()
+
+	a := agent.NewTestAgent(t, ``)
+	defer a.Shutdown()
+	testrpc.WaitForTestAgent(t, a.RPC, "dc1")
+
+	defaultCmdArgs := []string{
+		"-http-addr=" + a.HTTPAddr(),
+		"-token=root",
+	}
+	cases := []struct {
+		name           string
+		args           []string
+		expectedCode   int
+		errMsg         string
```

Comment:
Looks like we are not using `errMsg` variable, instead we need `output` and verify that the message returned is `group.verion.kind deleted`